### PR TITLE
made R.eq behave as Object.is from ES6

### PR DESCRIPTION
**Pull request by davidrusu**
Related to #724, almost a direct copy of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Polyfill

### CODE REVIEW

File: src/eq.js [hunk 1]
```diff
@@ -19,5 +19,20 @@ var _curry2 = require('./internal/_curry2');
...
+            return 1 / a === 1 / b;
+        }
+        if (a !== a) {
+            return b !== b;
+        }
+        return a === b;
+    } else {
+        return Object.is(a, b);
+    }
+});
```
Comment (davidchambers) [inline src/eq.js]: I don't see much value in using `Object.is` when available. Could the body of the function be an expression?

```javascript
a === 0 && b === 0 ? 1 / a === 1 / b : a === b || a !== a && b !== b
```

Comment (davidrusu) [inline src/eq.js]: Yup, it seems that they are equivalent, although I'm really not a big fan of the ternary operator, how about:
``` javascript
if (a === 0 && b === 0) {
    return 1 / a === 1 / b;
} else {
    return a === b || (a !== a && b != b);
}
```

Comment (davidchambers) [inline src/eq.js]: Works for me.

Comment (CrossEye) [inline src/eq.js]: When we first mentioned using `Object.is`, I passed around a somewhat different version of that ternary as a joke quiz: What does this function do?  Determines if two values are truly equal of course!  :smile: Gotta love JS.

File: src/eq.js [hunk 2]
```diff
@@ -19,5 +19,13 @@ var _curry2 = require('./internal/_curry2');
  *      R.eq(o, {}); //=> false
  *      R.eq(1, 1); //=> true
  *      R.eq(1, '1'); //=> false
+ *      R.eq(0, -0); //=> false
+ *      R.eq(NaN, NaN); //=> true
  */
-module.exports = _curry2(function eq(a, b) { return a === b; });
+module.exports = _curry2(function eq(a, b) {
+    if (a === 0 && b === 0) {
```
Comment (megawac) [inline src/eq.js]: nit: you only really need to check one or the other

File: src/eq.js [hunk 3]
```diff
@@ -19,5 +19,13 @@ var _curry2 = require('./internal/_curry2');
...
+        return 1 / a === 1 / b;
+    } else {
+        return a === b || (a !== a && b != b);
```
Comment (megawac) [inline src/eq.js]: strict for a and not strict for b? I'd go non strict for both here

Comment (davidrusu) [inline src/eq.js]: Oops!

**Comment by CrossEye** [comment]
@davidrusu:

I know that you've had more than your share of little nits to deal with in your recent check-ins, but do you want to fix these last ones so that we can merge this?

**Comment by davidrusu** [comment]
Sorry, It's been a busy past couple of days in school. Thanks for putting up with my beginner mistakes, I'm quite new to js and contributing in general.

**Comment by CrossEye** [comment]
No apologies necessary.  Thanks for your help.

This LGTM.  Anyone else have thoughts?

File: test/eq.js [hunk 4]
```diff
@@ -9,8 +9,12 @@ describe('eq', function() {
     it('tests for strict equality of its operands', function() {
         assert.strictEqual(R.eq(100, 100), true);
         assert.strictEqual(R.eq(100, '100'), false);
+        assert.strictEqual(R.eq('string', 'string'), true);
         assert.strictEqual(R.eq([], []), false);
         assert.strictEqual(R.eq(a, b), true);
+        assert.strictEqual(R.eq(0, -0), false);
+        assert.strictEqual(R.eq(NaN, NaN), true);
+        assert.strictEqual(R.eq(undefined, undefined), true);
```
Comment (davidchambers) [inline test/eq.js]: Let's add an assertion for `R.eq(null, undefined)`.

**Comment by davidchambers** [comment]
:deciduous_tree:

File: test/eq.js (hunk 4 above)
Comment (davidrusu) [inline test/eq.js]: added in new version

**Pull request merged**